src/hooks/use-scroll-percentage-analytics.ts

```diff
@@ -0,0 +1,70 @@
+import { useEffect } from 'react'
+import { canTrackAnalytics } from 'lib/analytics'
+
+const SCROLL_PERCENTAGE_THRESHOLDS = [25, 50, 75, 90]
+
+const getPercentageScrolled = (documentHeight, windowHeight) => {
+  const scrollOffset = window.scrollY + windowHeight
+
+  return Math.round(scrollOffset / documentHeight) * 100
+}
```

Comment:
Possibly related to percentages firing early: should `documentHeight` take `windowHeight` into account rather than scrollOffset?

For example, if window height is 1000, document height is 1200, scrollY of 20, expected 20 / 200 = 10% scrolled:

```js
const scrollOffset = window.scrollY + windowHeight // will be 1020
return Math.round(scrollOffset / documentHeight) * 100 // will be 1020 / 1200 = 0.85
```

I think this might be more correct:

```suggestion
const getPercentageScrolled = (documentHeight, windowHeight) => {
  const totalScrollableDistance = documentHeight - windowHeight
  const scrolledDistance = window.scrollY
  return Math.round(scrolledDistance / totalScrollableDistance) * 100
}
```

Reply:
ah right that makes total sense 🤦 i think the naming of the event threw me off - so if a visitor loads a 1200 pixel tall page in a 1000 pixel tall viewport, and hasn't scrolled at all, we'll report 75%. But that represents 75% of page content "seen" (or was visible), rather than 75% of the possible amount of scrolling they could do.